buildSrc/src/main/java/quilt/internal/tasks/setup/DownloadMinecraftJarsTask.java

```diff
@@ -7,85 +7,93 @@
 import com.google.common.hash.HashCode;
 import com.google.common.hash.Hashing;
 import com.google.common.io.Files;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.Internal;
 import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
+import org.quiltmc.launchermeta.version.v1.Downloads;
 import org.quiltmc.launchermeta.version.v1.Version;
 import quilt.internal.Constants;
 import quilt.internal.tasks.DefaultMappingsTask;
 
-public class DownloadMinecraftJarsTask extends DefaultMappingsTask {
+public abstract class DownloadMinecraftJarsTask extends DefaultMappingsTask {
     public static final String TASK_NAME = "downloadMinecraftJars";
-    @OutputFile
-    private final File clientJar;
+
+    @InputFile
+    public abstract RegularFileProperty getVersionFile();
+
+    @Internal("Fingerprinting is handled by getVersionFile")
```

Comment:
Once again, remember that conf cache is a thing -- if it's used at task run time, use something serializable instead; if it's not, then making it a transient field _should_ work.